wpilibc/src/main/native/include/frc/AsynchronousInterrupt.h

```diff
@@ -0,0 +1,73 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2020 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+#pragma once
+
+#include <frc/SynchronousInterrupt.h>
+
+#include <atomic>
+#include <memory>
+#include <thread>
+#include <utility>
+
+namespace frc {
+/**
+ * Class for handling sychronous interrupts.
+ *
+ * <p>Asynchronous interrupts are handled by the AsynchronousInterrupt class.
+ */
+class AsynchronousInterrupt {
+  explicit AsynchronousInterrupt(DigitalSource& source,
+                                 std::function<void(bool, bool)> callback);
+  explicit AsynchronousInterrupt(DigitalSource* source,
+                                 std::function<void(bool, bool)> callback);
+  explicit AsynchronousInterrupt(std::shared_ptr<DigitalSource> source,
+                                 std::function<void(bool, bool)> callback);
```

Comment:
It should be documented that the two arguments are for whether the callback should be called for rising or falling edges respectively.

Reply:
Also, these don't need to be explicit, as they have two args.